iron-list.html

```diff
@@ -486,19 +490,85 @@
         }
       }
 
-     // update the position of the items
+      // update the position of the items
       this._positionItems();
 
       // set the scroller size
       this._updateScrollerSize();
+
+      if (this._increasePoolIfNeeded()) {
+        // set models to the new items
+        this.async(this._update);
+      }
+    },
+
+    /**
+     * Creates a pool of DOM elements and attaches them to the local dom.
+     */
+    _createPool: function(size) {
+      var physicalItems = new Array(size);
+
+      this._ensureTemplatized();
+
+      for (var i = 0; i < size; i++) {
+        var inst = this.stamp(null);
+
+        // First element child is item; Safari doesn't support children[0]
+        // on a doc fragment
+        physicalItems[i] = inst.root.querySelector('*');
+        Polymer.dom(this).appendChild(inst.root);
+      }
+
+      return physicalItems;
     },
 
+    /**
+     * Increases the pool size. That is, the physical items in the DOM.
+     * This function will allocate twice as many physical items
+     * (limited by `MAX_PHYSICAL_COUNT`) if the content size is not at least
+     * three times the size of the viewport.
+     */
+    _increasePoolIfNeeded: function() {
+      if (this._physicalSize > this._viewportSize * 3) {
+        return false;
+      }
+
+      var nextPhysicalCount = Math.min(
+          // double the size of the physical items
+          this._physicalCount * 2,
+          this._virtualCount,
+          MAX_PHYSICAL_COUNT
+        );

```

Comment:
I think this gambit (try doubling every time) may be a little too aggressive.  Say you only needed 21 items, creating 40 would be unfortunate, and I can imagine people complaining about startup cost (people will make complex cards and things to put in the list where unnecessarily doubling how many we make will be measurable).

It would probably be better to keep something similar to this, but rather than doubling physical count, maybe do something like `nextPhysicalCount = physicalCount + (viewportSize*4 - physicalSize) / physicalAverage` (i.e. use the average to guess how many more are needed to meet the target 3x viewport, BUT actually target 4x the viewport to cover the case that the average is low so that you are unlikely to need to create more based on scrolling).... something like this?